tfjs-backend-webgpu/src/kernels/matmul_small_output_size_webgpu.ts

```diff
@@ -0,0 +1,201 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {backend_util, TensorInfo, util} from '@tensorflow/tfjs-core';
+import {WebGPUProgram} from './webgpu_program';
+
+export function makeMatMulSmallOutputSizeWidthSource(
+    aOuterSmall: boolean): string {
+  return `
+  float mm_readA(int row, int col);
+  float mm_readB(int row, int col);
+  void mm_write(int row, int col, float value);
+
+  const int TileAOuter = int(gl_WorkGroupSize.x);
+  const int TileBOuter = int(gl_WorkGroupSize.x);
+  const int TileInner = int(gl_WorkGroupSize.y - gl_WorkGroupSize.x);
+
+  shared float mm_Asub1[TileAOuter][TileInner];
+  shared float mm_Bsub1[TileInner][TileBOuter];
+  shared float mm_Asub2[TileAOuter][TileInner];
+  shared float mm_Bsub2[TileInner][TileBOuter];
+
+  void mm_matMul(int dimAOuter, int dimInner, int dimBOuter) {
+    int tileRowA = int(gl_LocalInvocationID.x);
```

Comment:
Please keep the same style with other matmul shader. Usually, we use gl_LocalInvocationID.x as col and gl_LocalInvocationID.y as row.